ArduPlane/stall_detection.cpp

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2020  Kraus-Hamdani Aerospace. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Plane.h"
+
+/*
+  stall detection algorithm and anti-stall management
+ */
+
+/*
+  Do we think we are stalled?
+  Probabilistic method where a bool is low-passed and considered a probability.
+*/
+void Plane::stall_detection_update(void)
+{
+    stall_state.raw_algorithm_output = false;
+
+    if (!arming.is_armed() ||
+        auto_state.last_flying_ms == 0 ||
+        !is_flying() ||
+        crash_state.is_crashed)
+    {
+        stall_state.stall_clear();
+        stall_state.count = 0;
+        stall_detection_log();
+        return;
+    }
+
+    if (stall_state.inhibit_ms != 0) {
+        // if someone told us to inihibt, freeze all stall processing for a brief time.
+        // This is useful when changing waypoints where we will likely perform a
```

Comment:
I'm open to suggestions to improve it

Reply:
We're no longer using this mechanism and I think it could be removed.

This was actually added to cope with predictable _roll_ errors arising due to `nav_roll` step changes on waypoint changes. This was fixed by calculating the rate limited nav roll.